src/capplet/mate-notification-applet.c

```diff
@@ -0,0 +1,218 @@
+{
+  gtk_widget_show_all (GTK_WIDGET (applet->applet));
+  if (active)
+    gtk_widget_hide (applet->image_on);
+  else
+    gtk_widget_hide (applet->image_off);
+}
+
+static void
+settings_changed (GSettings              *settings,
+                  gchar                  *key,
+                  MateNotificationApplet *applet)
+{
+  if (g_strcmp0 (GSETTINGS_KEY_DO_NOT_DISTURB, key) == 0)
+    set_status_image (applet,
+                      g_settings_get_boolean (settings, key));
+}
+
+static gboolean
+applet_main (MatePanelApplet *applet_widget)
+{
+  MateNotificationApplet *applet;
+  GtkWidget *box;
+
+  bindtextdomain (GETTEXT_PACKAGE, MATELOCALEDIR);
+  bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+  textdomain (GETTEXT_PACKAGE);
+
+  g_set_application_name (_("Do Not Disturb"));
+  gtk_window_set_default_icon_name ("mate-notification-properties");
+
+  applet = g_new (MateNotificationApplet, 1);
+  applet->applet = applet_widget;
+  applet->settings = g_settings_new (GSETTINGS_SCHEMA);
+
+  box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
+  applet->image_on  = gtk_image_new_from_icon_name ("user-available", GTK_ICON_SIZE_BUTTON);
+  applet->image_off = gtk_image_new_from_icon_name ("user-invisible", GTK_ICON_SIZE_BUTTON);
+
+  gtk_widget_set_tooltip_text (applet->image_off, N_("Do Not Disturb"));
+  gtk_widget_set_tooltip_text (applet->image_on, N_("Notifications Enabled"));
+
+  gtk_box_pack_start (GTK_BOX (box), applet->image_on,
+                      TRUE, TRUE, 0);
+  gtk_box_pack_start (GTK_BOX (box), applet->image_off,
+                      TRUE, TRUE, 0);
+  gtk_container_add (GTK_CONTAINER (applet_widget), box);
+
+  set_status_image (applet,
+                    g_settings_get_boolean (applet->settings,
+                                            GSETTINGS_KEY_DO_NOT_DISTURB));
+
+  /* set up context menu */
+  applet->action_group = gtk_action_group_new ("Do Not Disturb Actions");
+  gtk_action_group_set_translation_domain (applet->action_group, GETTEXT_PACKAGE);
+  gtk_action_group_add_actions (applet->action_group, applet_menu_actions,
+                                G_N_ELEMENTS (applet_menu_actions), applet);
+
+  GtkToggleAction *do_not_disturb_toggle_action =
+    gtk_toggle_action_new ("DoNotDisturb", N_("_Do not disturb"),
+                           N_("Enable/Disable do-not-disturb mode."), NULL);
+
+  gtk_action_group_add_action (applet->action_group,
+                               GTK_ACTION (do_not_disturb_toggle_action));
+
+  mate_panel_applet_setup_menu_from_resource  (applet->applet,
+                                               RESOURCE_PATH "menu.xml",
+                                               applet->action_group);
+
+  g_settings_bind (applet->settings, "do-not-disturb",
+                   do_not_disturb_toggle_action, "active",
+                   G_SETTINGS_BIND_DEFAULT);
+
+  g_signal_connect(G_OBJECT (applet->applet), "destroy",
```

Comment:
A space is missing before the bracket.
`g_signal_connect (G_OBJECT (applet->applet), "destroy",`